src/radical/pilot/agent/scheduler/base.py

```diff
@@ -840,6 +840,10 @@ def _try_allocation(self, unit):
         self._change_slot_states(slots, rpc.BUSY)
         unit['slots'] = slots
 
+        env_dict = unit['description'].get('environment', {})
```

Comment:
The task has always a description at this point, the UMGR should make sure of that.  If not, then this needs fixing.  So you can simplify the code to just setting the env).

Reply:
The unit does not always have an environment. I put this line to make sure that if there is an environment in the description to take it and if not to get an empty dictionary.

Do you think this is wrong?

Reply:
The default values will always be set, and that is an empty dictionary for the `environment` key:

```python
>>> import radical.pilot as rp
>>> cud = rp.ComputeUnitDescription({'executable': '/bin/true'})
>>> cud.environment
{}
```
We thus never need to check if a key was set by the application.